tests/test_resource_tracker.py

```diff
@@ -157,6 +159,65 @@ def create_resource(rtype):
                                                 errno.ENOENT))
         assert re.search(expected, err) is not None
 
+    @pytest.mark.parametrize("rtype", ["file", "folder", "semlock"])
+    def test_resource_tracker_refcounting(self, rtype):
+        if (sys.platform == "win32") and rtype == "semlock":
+            pytest.skip("no semlock on windows")
+
+        cmd = '''if 1:
+        import os
+        import tempfile
+        import time
+        from loky.backend.semlock import SemLock, _sem_open
+        from loky.backend import resource_tracker
+        from utils import resource_unlink, create_resource, resource_exists
+
+        resource_tracker.VERBOSE=True
+
+        try:
+            name = create_resource("{rtype}")
+            assert resource_exists(name, "{rtype}")
+
+            from loky.backend.resource_tracker import _resource_tracker
+            _resource_tracker.register(name, "{rtype}")
+            _resource_tracker.register(name, "{rtype}")
+
+            # Forget all information about the resource, but do not try to
+            # remove it
+            _resource_tracker.unregister(name, "{rtype}")
+            time.sleep(1)
+            assert resource_exists(name, "{rtype}")
+
+            _resource_tracker.register(name, "{rtype}")
+            _resource_tracker.register(name, "{rtype}")
+            _resource_tracker.maybe_unlink(name, "{rtype}")
+            time.sleep(1)
+            assert resource_exists(name, "{rtype}")
+
+            _resource_tracker.maybe_unlink(name, "{rtype}")
+            for _ in range(100):
+                if not resource_exists(name, "{rtype}"):
+                    break
+                time.sleep(.1)
+            else:
+                raise AssertionError("%s was not unlinked in time"  % name)
+        finally:
+            if resource_exists(name, "{rtype}"):
+                resource_unlink(name, "{rtype}")
+        '''
+
+        env = os.environ.copy()
+        env['PYTHONPATH'] = os.path.dirname(__file__)
+        p = subprocess.Popen(
+            [sys.executable, '-c', cmd.format(rtype=rtype)],
+            stderr=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            env=env
+        )
+        p.wait()
+        out, err = p.communicate()
+        assert p.returncode == 0, err
```

Comment:
If the test error, using this syntax instead of `check_output` leads to a much cleaner error reporting.